### PR TITLE
Addresses #36 Progress bar needs to be its own class

### PR DESCRIPTION
**Pull request by moseshll**

### CODE REVIEW

File: lib/progress_bar.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  def initialize(owner = '', options = {})
+    @done = 0
+    @steps = 1
+    @owner = owner
+    @options = options
+    @error = false
+    @newline = false
+  end
+
+  def step!(obj, action = '')
```
Comment (blancoj) [inline lib/progress_bar.rb:18]: @moseshll not sure what I think about the variable name obj. is this more like the present step?

File: lib/progress_bar.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    step! @steps, action
+  end
+
+  # The \033[K is to erase the entire line in case a previous action string
+  # might not be completely overwritten by a subsequent shorter one.
+  def draw(action = '')
+    return if @options[:no_progress] || @newline
+
+    progress = @steps.zero? ? 0 : 10 * @done / @steps
+    bar = format '%<bar>-10s', bar: block * progress
```
Comment (blancoj) [inline lib/progress_bar.rb:44]: @moseshll is this more like a progress_block, than a block.

File: lib/progress_bar.rb (hunk 1 above)
Comment (moseshll) [reply lib/progress_bar.rb:18]: It's just meant to be an integer. I think it's left over from my first looking at doing some kind of `def +()` operator. It's just meant to be the new integer that `@done` is set to. How would `  def step!(done, action = '')` look to you?

File: lib/progress_bar.rb (hunk 2 above)
Comment (blancoj) [reply lib/progress_bar.rb:44]: yes, I think that is better.

**Review by blancoj — APPROVED** [review]

**Pull request merged**